modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -200,12 +200,10 @@ const DataSet = ({
 	}
 
 	useEffect(() => {
-		const itemsAreInjected = !apiURL && itemsProp?.length !== items.length;
-
-		if (itemsAreInjected) {
+		if (itemsProp !== undefined) {
```

Comment:
We still have a check here, that may cause a bug later on. `itemsProp` is [set to `null` as a default prop](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js#L790), and `null` !== `undefined`. Please change this to a falsy check, `if (itemsProp) {`, or remove default prop, or both.

Reply:
@markocikos I made the change to a falsy check, I didn't remove the default prop because deploy fails as it identifies that `items` has already been declared